src/test/spec/unified_runner/entity.rs

```diff
@@ -0,0 +1,85 @@
+use crate::{bson::Bson, test::util::EventClient, Collection, Database};
+
+pub enum Entity {
+    Client(Client),
+    Database(Database),
+    Collection(Collection),
+    Result(Bson),
+    None,
+}
+
+pub struct Client {
+    client: EventClient,
+    observe_events: Option<Vec<String>>,
+    ignore_events: Option<Vec<String>>,
+}
+
+impl From<Database> for Entity {
+    fn from(database: Database) -> Self {
+        Self::Database(database)
+    }
+}
+
+impl From<Collection> for Entity {
+    fn from(collection: Collection) -> Self {
+        Self::Collection(collection)
+    }
+}
+
+impl From<Option<Bson>> for Entity {
+    fn from(result: Option<Bson>) -> Self {
+        match result {
+            Some(result) => Self::Result(result),
+            None => Self::None,
+        }
+    }
+}
+
+impl Entity {
+    pub fn from_client(
+        client: EventClient,
+        observe_events: Option<Vec<String>>,
+        ignore_events: Option<Vec<String>>,
+    ) -> Self {
+        Self::Client(Client {
+            client,
+            observe_events,
+            ignore_events,
+        })
+    }
+
+    pub fn as_client(&self) -> &EventClient {
```

Comment:
I think it would be more starightforward to return the `Client` type from above here. This would allow us to get rid of the below getters that are client specific. For ease of use, we could also implement `Deref<Target=EventClient>` on the `Client` type.

Reply:
Oh that's a cool trick, I originally did it this way because it seemed inconvenient to need to reference the inner field but deref made it work

Reply:
Yeah it's definitely a nifty feature. It's commonly used by pointer types like `Arc` to make them more ergonomic.